pipeline-runner/R/qc-helpers.R

```diff
@@ -216,6 +216,33 @@ getClusters <- function(clustering_method, resolution, data) {
   return(data)
 }
 
+#' Get and Convert SNN Graph object into igraph object
+#'
+#' This is used to facilitate leiden clustering.
+#'
+#' @param data \code{Seurat} object
+#'
+#' @return boolean indicating if SNN Graph object exists
+#'
+getSNNiGraph <- function(data, active.reduction) {
+  # check to see if we already have Seurat SNN Graph object
+  snn_name <- paste0(data@active.assay, "_snn")
+
+  # if doesn't exist, run SNN
+  if (!snn_name %in% names(data)) {
+    data <- Seurat::FindNeighbors(data, reduction = active.reduction)
+  }
+
+  # convert Seurat Graph object to igraph
+  # similar to https://github.com/joshpeters/westerlund/blob/46609a68855d64ed06f436a6e2628578248d3237/R/functions.R#L85
+  adj_matrix <-
+    Matrix::Matrix(as.matrix(data@graphs[[snn_name]]), sparse = TRUE)
+  g <- igraph::graph_from_adjacency_matrix(adj_matrix,
```

Comment:
in here, you can use better name for this "g".

Reply:
done, also renamed in worker